src/Workspaces/Core/Portable/Workspace/Solution/DocumentState.cs

```diff
@@ -534,7 +516,7 @@ private VersionStamp GetNewTreeVersionForUpdatedTree(SyntaxNode newRoot, Version
         // use static method so we don't capture references to this
         private static (ValueSource<TextAndVersion>, TreeAndVersion) CreateRecoverableTextAndTree(
             SyntaxNode newRoot,
-            string filePath,
+            string? filePath,
```

Comment:
So this is a bit funky. ISyntaxTreeFactoryService is annotated as allowing a null path, which makes this seem safe. But...the implementation isn't annotated. If you annotate it, you then discover the compiler API is annotated as not taking null. Except for:

https://github.com/dotnet/roslyn/blob/d5931e08dd388356176821aa25df094d0e9e94a8/src/Compilers/CSharp/Portable/Syntax/CSharpSyntaxTree.ParsedSyntaxTree.cs#L50

I just wrote https://github.com/dotnet/roslyn/pull/48820 to annotate enough to have the compiler flag the issue. I'll start a conversation whether we want to undo this bit or whether we want to change the compiler annotation to match the implementation.

Reply:
Oh, interesting. I though I saw the correct annotations in the usage.